tools/llzk-tblgen/CommonCAPIGen.cpp

```diff
@@ -0,0 +1,730 @@
+          // In raw token stream, these appear as a `raw_identifier` rather than `kw_*`.
+          if (tokens[j].is(tok::raw_identifier) && isAccessModifier(tokenText)) {
+            // If followed by a colon, skip that too
+            if (j + 1 < i && tokens[j + 1].is(tok::colon)) {
+              j++; // Skip the colon too
+            }
+            continue;
+          }
+
+          // Skip common implementation keywords that indicate we're in code, not a declaration.
+          // In raw token stream, this appears as a `raw_identifier` rather than `kw_return`.
+          if (tokens[j].is(tok::raw_identifier) && tokenText == "return") {
+            // This indicates we've hit implementation code, stop parsing
+            returnType.clear();
+            break;
+          }
+
+          // Skip modifiers and language keywords that shouldn't be in the return type
+          if (tokens[j].is(tok::raw_identifier) && isCppModifierKeyword(tokenText)) {
+            continue;
+          }
+
+          // Skip standalone colons (from lookback to access specifiers)
+          if (tokens[j].is(tok::colon)) {
+            // Only skip if it's not part of ::
+            if (j == 0 || j + 1 >= i || !tokens[j - 1].is(tok::colon)) {
+              continue;
+            }
+          }
+
+          // Add spacing between tokens (but not around ::)
+          if (!returnType.empty() && !returnType.ends_with("::") && tokenText != "::" &&
+              !tokenText.starts_with("::")) {
+            returnTypeStream << ' ';
+          }
+          returnTypeStream << tokenText;
+        }
+        // Trim possible whitespace
+        returnType = StringRef(returnType).trim().str();
+      }
+
+      // Find matching ')' and check for parameters
+      size_t closeParenIdx = tokenCount;
+      bool hasParameters = false;
+      std::vector<MethodParameter> parameters;
+      {
+        // Initialize parenDepth to 1 to account for the opening '(' at tokens[i+1]
+        // Start scanning from i+2 (the first token after the opening paren)
+        size_t parenDepth = 1;
+        for (size_t j = i + 2; j < tokenCount; ++j) {
+          if (tokens[j].is(tok::l_paren)) {
+            parenDepth++;
+          } else if (tokens[j].is(tok::r_paren)) {
+            parenDepth--;
+            if (parenDepth == 0) {
+              closeParenIdx = j;
+
+              // Parse parameters between '(' and ')'
+              // Parameters follow the pattern: type name [, type name ...]
+              const std::vector<Token> paramTokens = [&tokens, i, j, tokenCount]() {
+                std::vector<Token> temp;
+                for (size_t k = i + 2; k < j; ++k) {
+                  if (k >= tokenCount) {
+                    break;
+                  }
+                  if (!tokens[k].is(tok::comment)) {
+                    temp.push_back(tokens[k]);
+                  }
+                }
+                return temp;
+              }();
```

Comment:
```suggestion
              std::vector<Token> paramTokens;
              for (size_t k = i + 2; k < j; ++k) {
                if (k >= tokenCount) {
                  break;
                }
                if (!tokens[k].is(tok::comment)) {
                  paramTokens.push_back(tokens[k]);
                }
              }
```
The single-use lambda seems a bit unnecessary.